.pre-commit-config.yaml

```diff
@@ -0,0 +1,31 @@
+repos:
+  - repo: https://github.com/psf/black
+    rev: 22.3.0
+    hooks:
+    - id: black
+      language_version: python3
+  - repo: https://github.com/PyCQA/flake8
+    rev: 4.0.1
+    hooks:
+      - id: flake8
+        additional_dependencies: [flake8-eradicate==1.2.0]
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.2.0
+    hooks:
+      - id: check-toml
+      - id: check-yaml
```

Comment:
It would be worth auditing [the full list of available hooks](https://pre-commit.com/hooks.html) to see if there's anything else useful there. Other projects (like [pyscript](https://github.com/pyscript/pyscript/blob/main/.pre-commit-config.yaml) or [pip](https://github.com/pypa/pip/blob/main/.pre-commit-config.yaml)) may be a good reference here.

Reply:
Added a few more for trailing whitespace, newline for end-of-file, etc. :+1:

Reply:
One additional one that would be good would be to add mypy as a pre-commit hook, even if we aren't planning on or planning on adding incremental type hints. It's analysis found errors, and we could resolve these and make it part of a future PR.